Makefile.rules

```diff
@@ -984,26 +984,25 @@ testbugfix: all
           ReadGapRoot( "tst/testbugfix.g" );' | $(TESTGAP) | \
             tee `date -u +dev/log/testbugfix2_%Y-%m-%d-%H-%M` )
 
+# run a test in tst/testlibgap
 tst/testlibgap/%: obj/tst/testlibgap/%.lo obj/tst/testlibgap/common.lo libgap.la
-	$(QUIET_LINK)$(LINK) $(GAP_LDFLAGS) $^ -o $@
+	$(LINK) $(GAP_LDFLAGS) $^ -o $@
+	$@ -A -l $(top_srcdir) -m 32m -q -T --nointeract > $@.out
+	diff $(top_srcdir)/$@.expect $@.out
```

Comment:
I disagree with this part of the change: IMHO, the build rules for the test executables should be separate from the (PHONY) rules that run the tests.

Indeed, right now these targets are *not* marked as .PHONY, so I expect (but did not try) that running `make testlibgap` twice will not actually run any tests the second time around.

One way to do this should be to remove the above two lines here again; and instead have a for loop in `testlibgap` which iterates over `$^` (resp. `tst/testlibgap/basic tst/testlibgap/wscreate tst/testlibgap/wsload`) and for each executes the above code. That way, you also automatically get the order right.

Reply:
fixed in the latest commit